<?xml version="1.0"?>

<launch>
  <param name="/robot_description_semantic" textfile="$(find thor_mang_moveit_config)/config/thormang3.srdf" />
  
  <node pkg="message_to_tf" type="message_to_tf" name="imu_to_tf" output="screen">
    <!--<param name="frame_id" value="base_footprint" />-->
    <!--<param name="footprint_frame_id" value="base_footprint2" />-->
    <param name="stabilized_frame_id" value="pelvis_stabilized" />
    <param name="child_frame_id" value="pelvis_link" />
    <param name="topic" value="/state_estimator/pelvis_pose" />
    <!--<param name="tf_prefix" value="ground_truth" />-->
  </node>

  <node pkg="message_to_tf" type="message_to_tf" name="foot_frame_to_tf" output="screen">
    <!--<param name="frame_id" value="world" />-->
    <param name="footprint_frame_id" value="footstep_base_" />
    <param name="stabilized_frame_id" value="footstep_stabilized" />
    <param name="child_frame_id" value="footstep_frame" />
    <param name="topic" value="/state_estimator/ground_point" />
  </node>

  <node name="state_estimator" pkg="biped_state_estimator" type="state_estimator_test" output="screen"> 
    <remap from="~pelvis_imu" to="/johnny5/sensor/imu/filtered"/>
    <remap from="~joint_states"         to="/johnny5/joints/joint_states"/>
    <param name="pelvis_name"           value="pelvis_link"/>
    <param name="right_foot_name"       value="r_leg_foot_link"/>
    <param name="left_foot_name"        value="l_leg_foot_link"/>
  </node>
  
</launch>
